utils/sampler/lib/helper_widgets.dart

```diff
@@ -0,0 +1,354 @@
+        assert(file == null || file.absolute.path.contains(location.absolute.path),
+            'Supplied file must be within location directory'),
+        super(key: key);
+
+  final Directory location;
+  final File? file;
+  final String label;
+  final String _fileBrowserName;
+  final int startLine;
+
+  static String _getFileBrowserName(Platform platform) {
+    switch (platform.operatingSystem) {
+      case 'windows':
+        return 'FILE EXPLORER';
+      case 'macos':
+        return 'FINDER';
+      case 'linux':
+      default:
+        return 'FILE BROWSER';
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final TextStyle labelStyle = Theme.of(context).textTheme.bodyText2!;
+    final String path = file?.path ?? location.absolute.path;
+    return DefaultTextStyle(
+      style: labelStyle,
+      child: Padding(
+        padding: const EdgeInsets.all(4.0),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.end,
+          children: <Widget>[
+            Row(
+              mainAxisSize: MainAxisSize.min,
+              crossAxisAlignment: CrossAxisAlignment.center,
+              children: <Widget>[
+                Padding(
+                  padding: const EdgeInsetsDirectional.only(start: 8.0, end: 8.0),
+                  child: Text(
+                    '$label${label.isNotEmpty ? ' ' : ''}$path',
+                    overflow: TextOverflow.ellipsis,
+                  ),
+                ),
+                Padding(
+                  padding: const EdgeInsetsDirectional.only(start: 8.0, end: 8.0),
+                  child: IconButton(
+                    tooltip: 'Copy path to clipboard',
+                    icon: const Icon(Icons.copy),
+                    onPressed: () {
+                      Clipboard.setData(ClipboardData(text: path));
+                    },
+                  ),
+                ),
+              ],
+            ),
+            Row(
+              mainAxisSize: MainAxisSize.min,
+              crossAxisAlignment: CrossAxisAlignment.center,
+              children: <Widget>[
+                Padding(
+                  padding: const EdgeInsetsDirectional.only(start: 8.0, end: 8.0),
+                  child: TextButton(
+                    child: Text('OPEN IN $_fileBrowserName'),
+                    onPressed: () {
+                      openFileBrowser(file ?? location);
+                    },
+                  ),
+                ),
+                for (final IdeType type in IdeType.values)
+                  Padding(
+                    padding: const EdgeInsetsDirectional.only(start: 8.0, end: 8.0),
+                    child: TextButton(
+                      child: Text('OPEN IN ${getIdeName(type).toUpperCase()}'),
+                      onPressed: () {
+                        openInIde(type, location, file: file, startLine: startLine);
+                      },
+                    ),
+                  ),
```

Comment:
Do these 'OPEN IN' features work on macOS without signing? If not, we might want to either hide these or grey them out with some way to get the path so we can copy it into an editor or finder to access the extracted file.

Reply:
Yes, they do work without signing: I have the sandbox turned off when you build it. But there is already a "copy to clipboard" button for the path (and it's selectable text too), because not everyone uses IntelliJ or VSCode.